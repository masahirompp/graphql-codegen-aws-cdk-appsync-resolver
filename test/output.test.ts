import { SynthUtils } from "@aws-cdk/assert";
import { App } from "@aws-cdk/core";
import { TestStack } from "./TestStack";

test("CloudFormation Test", () => {
  // prepare
  const stack = new TestStack(new App(), "TestStack");

  // test 1
  expect(
    SynthUtils.toCloudFormation(stack)["Resources"][
      "AppSyncResolverQueryGetUser"
    ]
  ).toMatchSnapshot();

  // test 2
  expect(
    SynthUtils.toCloudFormation(stack)["Resources"][
      "TestAppSyncResolverQueryListUsers"
    ]
  ).toMatchSnapshot();
});
