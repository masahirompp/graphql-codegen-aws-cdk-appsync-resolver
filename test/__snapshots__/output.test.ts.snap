// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloudFormation Test 1`] = `
Object {
  "DependsOn": Array [
    "DataSourceQueryGetUser",
    "ItemsApiSchemaDC216D83",
  ],
  "Properties": Object {
    "ApiId": Object {
      "Fn::GetAtt": Array [
        "ItemsApi3788C05C",
        "ApiId",
      ],
    },
    "DataSourceName": "DataSourceQueryGetUser",
    "FieldName": "getUser",
    "TypeName": "Query",
  },
  "Type": "AWS::AppSync::Resolver",
}
`;

exports[`CloudFormation Test 2`] = `
Object {
  "DependsOn": Array [
    "DataSourceQueryListUsers",
    "ItemsApiSchemaDC216D83",
  ],
  "Properties": Object {
    "ApiId": Object {
      "Fn::GetAtt": Array [
        "ItemsApi3788C05C",
        "ApiId",
      ],
    },
    "DataSourceName": "DataSourceQueryListUsers",
    "FieldName": "listUsers",
    "TypeName": "Query",
  },
  "Type": "AWS::AppSync::Resolver",
}
`;
